src/nbsap/models.py

```diff
@@ -347,9 +348,40 @@ def get_next_code(self):
                 letter = 'a'
             return self.parent.code + letter
 
-        codes = [int(RE_ACTION_CODE.match(code).groups()[0]) for code in
-                 EuAction.objects.values_list('code', flat=True)]
-        return str(max(codes) + 1)
+        #action code generation
+        all_actions = EuAction.objects.all()
```

Comment:
You should do some sorting at this stage. `.all()` usually sorts entries by pk/id/date inserted, and this may not be the ordering you want for renumbering them.